src/utils/spam_detection.py

```diff
@@ -0,0 +1,115 @@
+]
+
+
+# Function to calculate normalised Levenshtein distance and detect spam
+def is_spam(input_message, spam_messages, threshold=0.2):
+    """
+    Detects if the input message is similar to known spam messages based on Levenshtein distance.
+    The distance is normalised, so it ranges from 0 (exact match) to 1 (completely different).
+
+    Args:
+    - input_message (str): The message to classify.
+    - spam_messages (list): A list of known spam messages.
+    - threshold (float): The threshold below which the message is considered spam.
+
+    Returns:
+    - bool: Indicates if the message is spam.
+    """
+    for spam_message in spam_messages:
+        # Calculate the Levenshtein distance between the input and the spam message
+        distance = Levenshtein.distance(input_message.lower(), spam_message.lower())
+        max_len = max(len(input_message), len(spam_message))
+        normalised_distance = distance / max_len
+
+        # If the Levenshtein distance is below the threshold, classify as spam
+        if normalised_distance < threshold:
+            return True
+
+    # If no message is close enough to be considered spam, classify as not spam
+    return False
+
+
+async def check_spam(message):
+    """
+    Checks potential spam messages by deleting them and timing out the user.
+
+    Args:
+    - message (discord.Message): The message object to evaluate.
+    """
+    input_message = message.content
+    is_spam_flag = is_spam(input_message, spam_messages)
+
+    # If the message is spam, take action
+    if is_spam_flag:
+        try:
+            # Try to delete the spam message
+            await message.delete()
+        except Exception as e:
+            print(f"An error occurred: {e}")
+
+        member = message.author
+
+        # Check if the bot can timeout this member
+        bot_role = message.guild.me.top_role
+        member_top_role = member.top_role
+
+        if member_top_role >= bot_role:
+            print(
+                f"Cannot timeout {member.display_name}: Their role is higher or equal to the bot's role."
+            )
+            return
+
+        try:
+            # Timeout the user for 1 day
+            await member.timeout(
+                datetime.timedelta(days=1), reason="Sending spam messages"
+            )
+            print(f"User {member} has been timed out for 1 day.")
+        except Exception as e:
+            print(f"An error occurred: {e}")
+
+        # Log the spam message
+        try:
+            log_channel = message.guild.get_channel(LOG_CHANNEL_ID)
+
+            # Create an embed to log the spam message
+            embed = discord.Embed(
+                description=f"**Message sent by {member.mention} in {message.channel.mention} was flagged as spam, deleted, and the user has been timed out for 1 day. Review the message and ban the member if confirmed as spam.**",
```

Comment:
```suggestion
                description=f"**Message sent by {member.mention} in {message.channel.mention} was flagged as spam, deleted, and the user has been timed out for 1 day. Review the message and take appropriate action if confirmed as spam.**",
```

In most cases we'd probably ban them but banning may not always be the policy. We should keep it more general